src/js/inverted-index.js

```diff
@@ -40,54 +40,61 @@ var Index = function () {
    */
   this.createIndex = function (fileName, fileContents) {
 
-    /** This will contain the indexed file contents */
-    var indexedFileContents = {
-      indexMap: {}
-    };
-
-    /** Gets index number of each document in the JSON object */
-    indexedFileContents.docIndexNum = fileContents.map(function (item, indexNum) {
-      return indexNum;
-    });
+    /**
+     * Ensure that the uploaded JSON file is valid
+     */
+    try {
+      /** This will contain the indexed file contents */
+      var indexedFileContents = {
+        indexMap: {}
+      };
 
-    fileContents.forEach(function (item, indexNum) {
+      /** Gets index number of each document in the JSON object */
+      indexedFileContents.docIndexNum = fileContents.map(function (item, indexNum) {
+        return indexNum;
+      });
 
-      var titleTokens;
-      var textTokens;
+      fileContents.forEach(function (item, indexNum) {
 
-      /**
-       * Check if each document in the JSON file
-       * has a title and text.
-       */
-      if (item.title && item.text) {
-        /** Tokenize both title and text */
-        titleTokens = this.tokenize(item.title);
-        textTokens = this.tokenize(item.text);
-      } else {
-        return alert("Document " + indexNum + ":  Invalid JSON document. It should have text and title");
-      }
+        var titleTokens;
+        var textTokens;
 
+        /**
+         * Check if each document in the JSON file
+         * has a title and text.
+         */
+        if (item.title && item.text) {
+          /** Tokenize both title and text */
+          titleTokens = this.tokenize(item.title);
+          textTokens = this.tokenize(item.text);
+        } else {
+          return alert("Document " + indexNum + " should have text and title");
```

Comment:
## Critical

You should never, not any under circumstance return an alert, instead throw an exception